### PR TITLE
File sync for envd-server

### PR DESCRIPTION
**Pull request by AlexXi19**
Draft pull request for https://github.com/tensorchord/envd-server/issues/102

### CODE REVIEW

File: pkg/syncthing/config.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+			Enabled:    true,
+			RawAddress: "0.0.0.0:8384",
+			APIKey:     "envd",
+			Theme:      "default",
+		},
+		Options: config.OptionsConfiguration{
+			GlobalAnnEnabled:     false,
+			LocalAnnEnabled:      false,
+			ReconnectIntervalS:   1,
+			StartBrowser:         true, // TODO: disable later
```
Comment (VoVAllen) [inline pkg/syncthing/config.go:23]: fix TODO

File: pkg/syncthing/config.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			NATEnabled:           false,
+			URAccepted:           1,
```
Comment (VoVAllen) [inline pkg/syncthing/config.go:25]: Let's make this -1 to avoid sending anonymous usage data

**Review by VoVAllen — COMMENTED** [review]
Overall looks good to me. Thanks for your contribution! 

Further step:
- fix lints
- Add options in environment create API endpoint for the user to choose whether provision synching sidecar or not.

File: pkg/syncthing/config.go (hunk 1 above)
Comment (AlexXi19) [reply pkg/syncthing/config.go:23]: Ok, I'm only 70% done with the functionality, will do a final clean up later.

**Pull request closed without merging**